guide/interactions/autocomplete.md

```diff
@@ -0,0 +1,110 @@
+# Autocomplete
+
+Slash command options provide the feature of autocomplete. In this section we'll cover how to both add autocomplete support to your command and also how to respond to it.
+
+::: tip
+This page is a follow-up to the [interactions (slash commands) pages](/interactions/registering-slash-commands.md). Please carefully read those first so that you can understand the methods used in this section.
+:::
+
+::: warning
+Make sure that you're on discord.js version 13.3.0 or above before using autocomplete.
+:::
+
+## Preparing commands
+
+To use autocomplete with your command options, you have to tell Discord to enable it.
+How to do this is dependent on how you create your slash commands.
+
+### Slash command builder
+
+The slash command builder does not yet support using autocomplete.
+
+### Custom system
+
+For other custom systems or entering it raw, your resulting code should look something like this. To enable autocomplete, set it to true in the option.
+
+```js {10}
+const commandData = {
+	name: 'autocomplete',
+	description: 'test command to show how autocomplete should be setup',
+	defaultPermission: true,
+	options: [
+		{
+			name: 'name',
+			description: 'Name of something',
+			type: 'STRING',
```

Comment:
For this custom system, the `type: STRING` only works for when the client is logged in and this data is passed to `<Guild>.commands?.create()` (or other methods that take ApplicationCommandData). As all of this guide uses REST to register commands, shouldn't this be changed to the corresponding integer value `type: 3`?

Reply:
I saw your discussion in discord, and your suggestion has a really good point, and using `type: 3` works with both and should thus be prefered. However, I'd suggest writing it as `type: 3, // String` to clarify.

Reply:
I think that would work. Is it just the string option that is able to use autocomplete, or are there others too?

Reply:
I think it's Integers as well, not 100% sure tho. And to clarify, my suggestion was just to add a comment at the end of `type: 3` to indicate that 3 means string.

Reply:
Just tested that out. `integer` and `number` types can be used for autocomplete, 4 and 10 respectively. The others are basically autocompleted options.

Reply:
Please change your `type: 3,` to `type: 3, // String` before I can commit it.

Reply:
> Please change your `type: 3,` to `type: 3, // String` before I can commit it.

👍